doc/modules/clustering.rst

```diff
@@ -545,26 +546,26 @@ graph, and SpectralClustering is initialized with `affinity='precomputed'`::
 
 .. topic:: References:
 
- * `"A Tutorial on Spectral Clustering"
-   <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.165.9323>`_
+ * :doi:`"A Tutorial on Spectral Clustering"
    Ulrike von Luxburg, 2007
+   <10.1007/s11222-007-9033-z>`
 
- * `"Normalized cuts and image segmentation"
-   <http://citeseer.ist.psu.edu/viewdoc/summary?doi=10.1.1.160.2324>`_
+ * :doi:`"Normalized cuts and image segmentation"
    Jianbo Shi, Jitendra Malik, 2000
+   <10.1109/34.868688>`
 
- * `"A Random Walks View of Spectral Segmentation"
-   <http://citeseer.ist.psu.edu/viewdoc/summary?doi=10.1.1.33.1501>`_
+ * "A Random Walks View of Spectral Segmentation"
    Marina Meila, Jianbo Shi, 2001
+   <http://citeseer.ist.psu.edu/viewdoc/summary?doi=10.1.1.33.1501>`
 
- * `"On Spectral Clustering: Analysis and an algorithm"
-   <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.19.8100>`_
+ * :doi:`"On Spectral Clustering: Analysis and an algorithm"
    Andrew Y. Ng, Michael I. Jordan, Yair Weiss, 2001
+   <http://citeseerx.ist.psu.edu/viewdoc/summary?doi=10.1.1.19.8100>`
```

Comment:
I can't find a DOI for this one, please revert, as this can't be anything else than an external link